dev/diffs/3.5.6.diff

```diff
@@ -1938,7 +1938,17 @@ index 8e88049f51e..d3c0737d52e 100644
      import testImplicits._
      // keep() should take effect on StartsWith/EndsWith/Contains
      Seq(
-@@ -1984,7 +1997,8 @@ abstract class ParquetFilterSuite extends QueryTest with ParquetTest with Shared
+@@ -1743,7 +1756,8 @@ abstract class ParquetFilterSuite extends QueryTest with ParquetTest with Shared
+     }
+   }
+ 
+-  test("SPARK-17091: Convert IN predicate to Parquet filter push-down") {
++  test("SPARK-17091: Convert IN predicate to Parquet filter push-down",
++    IgnoreCometNativeScan("Comet has different push-down behavior")) {
```

Comment:
So this has been working for regular Comet scan?

Reply:
Yes, this works for `native_comet` scan, so we are only ignoring it for the new native scans

Reply:
Does that mean we need to revisit to support parquet filter push-down? If so, should we track in a ticket?

Reply:
I believe we don't need to revisit predicate push-down in general. This specific case  is only for the predicate involving an IN clause 
So in `select a from foo where b = 2` the predicate will be pushed down
but in `select a from foo where b in (2)` the predicate is not being pushed down it appears
We have not investigate this failure too deeply.

Reply:
> But it says `native_comet` is working for this? Should we investigate why it is not working for `native_iceberg_compat`? @andygrove @parthchandra

I'll take a look at this.

Reply:
Tracked https://github.com/apache/datafusion-comet/issues/1940